thrall/app/lib/kinesis/MessageProcessor.scala

```diff
@@ -51,14 +53,19 @@ class MessageProcessor(es: ElasticSearch,
     Future.sequence(es.bulkInsert(imagesToIndex))
   }
 
-  def updateImageUsages(message: UpdateMessage)(implicit ec: ExecutionContext) = {
+  def updateImageUsages(message: UpdateMessage)(implicit ec: ExecutionContext, mc: MarkerContext): Future[List[ElasticSearchUpdateResponse]] = {
     implicit val unw = Json.writes[UsageNotice]
+
     def asJsLookup(us: Seq[Usage]): JsLookupResult = JsDefined(Json.toJson(us))
+
     withId(message) { id =>
       withUsageNotice(message) { usageNotice =>
         withLastModified(message) { lastModifed =>
           val usages = usageNotice.usageJson.as[Seq[Usage]]
-          Future.sequence(es.updateImageUsages(id, asJsLookup(usages), dateTimeAsJsLookup(lastModifed)))
+          //DANGER: Here be an entirely useless sequence and map
+          Future.sequence(es.updateImageUsages(id, asJsLookup(usages), dateTimeAsJsLookup(lastModifed)).map(_.recoverWith {
+            case ElasticNotFoundException => Future.successful(ElasticSearchUpdateResponse())
+          }))
```

Comment:
This would be better expressed with a `traverse`

Reply:
thanks :)